app/src/main/java/org/simple/clinic/util/room/MapRoomTypeConverter.kt

```diff
@@ -0,0 +1,28 @@
+package org.simple.clinic.util.room
+
+import androidx.room.TypeConverter
+import java.util.TreeMap
+import javax.inject.Inject
+
+class MapRoomTypeConverter @Inject constructor() {
+  @TypeConverter
+  fun fromMap(value: Map<String, String>): String {
+    val sortedMap = TreeMap(value)
+    return sortedMap.keys.joinToString(separator = ",").plus("<divider>")
+        .plus(sortedMap.values.joinToString(separator = ","))
```

Comment:
just using `,` for a separator is a bad idea. Value might contain a comma, making conversion of string to map impossible.
Can you check Java/Kotlin methods on converting Map to String?

Reply:
GSON also has map->string and string->map conversion logic. Check that out as well.
<img width="738" alt="image" src="https://user-images.githubusercontent.com/9414539/218029557-4bb0549a-06fc-4851-8c3a-eaec4750b457.png">